### PR TITLE
Revert "Pin urllib3 to <2.0.0 for now"

### LINKED ISSUE #2070: Support urllib3 2.0.0
### Problem Statement

see if we need any changes and how the openssl support matrix works across runtimes (AWS lambdas etc)
https://urllib3.readthedocs.io/en/latest/v2-migration-guide.html

### PR DESCRIPTION
**Pull request by asottile-sentry**
Reverts getsentry/sentry-python#2069

Resolves #2070

this shouldn't have been pinned in the first place.  the actual failure was unrelated to sentry.  _they_ should have pinned in their app -- a library should not make this decision for its users

### CODE REVIEW

**Comment by sl0thentr0py** [comment]
there is more complexity to the discussion and an open PR at https://github.com/getsentry/sentry-python/pull/2137
I was OOO, we will release this soon, closing.

**Comment by asottile-sentry** [comment]
that PR is based on an incorrect assumption and does not include the necessary fixes to typing, docs, etc.

**Review by sentrivana — CHANGES_REQUESTED** [review]
See comment, otherwise LGTM.

File: sentry_sdk/transport.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def record_lost_event(
         self._discarded_events[data_category, reason] += quantity
 
     def _update_rate_limits(self, response):
-        # type: (urllib3.HTTPResponse) -> None
+        # type: (urllib3.BaseHTTPResponse) -> None
```
[the remark this answers is not in the document]
Comment (asottile-sentry) [reply sentry_sdk/transport.py:189]: this is only used in a type comment and has no effect on the runtime nor compatibility

[the remark this answers is not in the document]
Comment (asottile-sentry) [reply sentry_sdk/transport.py:189]: additionally the type checker will not understand the `try` / `except` pattern:

```
pip install mypy types-urllib3 'urllib3<2'
```

```console
$ cat t.py 
try:
    from urllib3 import BaseHTTPResponse as Response
except ImportError:
    from urllib3 import HTTPResponse as Response


reveal_type(Response)
$ mypy t.py
t.py:2: error: Module "urllib3" has no attribute "BaseHTTPResponse"; maybe "HTTPResponse"?  [attr-defined]
t.py:7: note: Revealed type is "Any"
Found 1 error in 1 file (checked 1 source file)
```

**Review by sentrivana — APPROVED** [review]

**Pull request merged**